### PR TITLE
mgmt, authorization, support description in role assignment

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
Fix https://github.com/Azure/azure-sdk-for-java/issues/31421

The "condition" is still in preview, it complicates the efforts to upgrade 2022-04-01.

Maybe we need to use combination of 2022-04-01 with role assignment 2020-10-01-preview

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-authorization/src/test/java/com/azure/resourcemanager/authorization/RoleAssignmentTests.java [hunk 1]
```diff
@@ -46,6 +47,7 @@ public void canCRUDRoleAssignment() throws Exception {
             Assertions.assertEquals(roleAssignment.scope(), roleAssignment1.scope());
             Assertions.assertEquals(roleAssignment.roleDefinitionId(), roleAssignment1.roleDefinitionId());
             Assertions.assertEquals(roleAssignment.principalId(), roleAssignment1.principalId());
+            Assertions.assertEquals("contributor role", roleAssignment1.description());
```
[the remark this answers is not in the document]
Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-authorization/src/test/java/com/azure/resourcemanager/authorization/RoleAssignmentTests.java:50]: So far prefer to use the value I know is correct.

If we had bug that description in both roleAssignment and roleAssignment1 is null, changed assert will still pass.

File: sdk/resourcemanager/azure-resourcemanager-authorization/src/main/java/com/azure/resourcemanager/authorization/models/RoleAssignmentProperties.java [hunk 2]
```diff
@@ -25,6 +25,9 @@ public final class RoleAssignmentProperties {
     @JsonProperty(value = "principalId", required = true)
     private String principalId;
 
+    @JsonProperty(value = "description")
+    private String description;
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-authorization/src/main/java/com/azure/resourcemanager/authorization/models/RoleAssignmentProperties.java:29]: This is through code regen?

**Review by XiaofeiCao — APPROVED** [review]
LGTM

File: sdk/resourcemanager/azure-resourcemanager-authorization/src/main/java/com/azure/resourcemanager/authorization/models/RoleAssignmentProperties.java (hunk 2 above)
Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-authorization/src/main/java/com/azure/resourcemanager/authorization/models/RoleAssignmentProperties.java:29]: No. Currently this part is kind of hybrid of 2015-07-01 and 2020-10-01-preview (based on 2015-07-01, but add "condition" and switched api-version in API call to 2020-10-01-preview).

We probably going to regen it with 2020-10-01-preview, when requested for 2022-04-01 ("description" in 2022-04-01)

**Review by haolingdong-msft — APPROVED** [review]

File: sdk/resourcemanager/azure-resourcemanager-authorization/src/main/java/com/azure/resourcemanager/authorization/models/RoleAssignmentProperties.java (hunk 2 above)
Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-authorization/src/main/java/com/azure/resourcemanager/authorization/models/RoleAssignmentProperties.java:29]: Seems they already have "condition" in their stable swagger?
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/authorization/resource-manager/Microsoft.Authorization/stable/2022-04-01/authorization-RoleAssignmentsCalls.json#L594-L597

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-authorization/src/main/java/com/azure/resourcemanager/authorization/models/RoleAssignmentProperties.java:29]: OK, that would be great.

I thought I've tried the codegen with 2022-04-01...
But we will still wait for the request for a while, from .NET they've got report that some region was not available in 2022-04-01

Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-authorization/src/main/java/com/azure/resourcemanager/authorization/models/RoleAssignmentProperties.java:29]: ok

**Pull request merged**